src/OSPSuite.Core/Domain/Module.cs

```diff
@@ -9,28 +10,28 @@
 
 namespace OSPSuite.Core.Domain
 {
-   public class Module : ObjectBase
+   public class Module : ObjectBase, IEnumerable<IBuildingBlock>
    {
-      private readonly List<MoleculeStartValuesBuildingBlock> _moleculeStartValuesCollection = new List<MoleculeStartValuesBuildingBlock>();
-      private readonly List<ParameterStartValuesBuildingBlock> _parameterStartValuesCollection = new List<ParameterStartValuesBuildingBlock>();
+      private readonly List<IBuildingBlock> _buildingBlocks = new List<IBuildingBlock>();
+
+      private T buildingBlockByType<T>() where T : IBuildingBlock => _buildingBlocks.OfType<T>().SingleOrDefault();
+      private IReadOnlyList<T> buildingBlocksByType<T>() where T : IBuildingBlock => _buildingBlocks.OfType<T>().ToList();
 
       public bool ReadOnly { get; set; } = false;
 
-      public EventGroupBuildingBlock EventGroups { set; get; }
-      public MoleculeBuildingBlock Molecules { set; get; }
-      public ObserverBuildingBlock Observers { set; get; }
-      public ReactionBuildingBlock Reactions { set; get; }
-      public PassiveTransportBuildingBlock PassiveTransports { set; get; }
-      public SpatialStructure SpatialStructure { set; get; }
-      public IReadOnlyList<MoleculeStartValuesBuildingBlock> MoleculeStartValuesCollection => _moleculeStartValuesCollection;
-      public IReadOnlyList<ParameterStartValuesBuildingBlock> ParameterStartValuesCollection => _parameterStartValuesCollection;
+      public EventGroupBuildingBlock EventGroups => buildingBlockByType<EventGroupBuildingBlock>();
```

Comment:
Still adding those accessor props so that the code is mostly unchanged. But how we manage the list internally has changed